adapters/sharethrough/utils.go

```diff
@@ -183,17 +193,23 @@ func (u Util) gdprApplies(request *openrtb.BidRequest) bool {
 	return gdprApplies != 0
 }
 
-func (u Util) gdprConsentString(request *openrtb.BidRequest) string {
-	var consentString string
-
-	if request.User != nil {
-		if jsonExtUser, err := request.User.Ext.MarshalJSON(); err == nil {
-			// empty string is the return value if error, so no need to handle
-			consentString, _ = jsonparser.GetString(jsonExtUser, "consent")
+func (u Util) parseUserExt(user *openrtb.User) userInfo {
+	var userExt userExt
+	var unifiedId string
+	var gdprConsentString string
+	if user != nil && user.Ext != nil {
+		if err := json.Unmarshal(user.Ext, &userExt); err == nil {
+			gdprConsentString = userExt.Consent
+			for i := 0; i < len(userExt.Eids); i++ {
+				if userExt.Eids[i].Source == "adserver.org" {
+					unifiedId = userExt.Eids[i].Uids[0].ID
```

Comment:
Would be good to check if `Eids[i].Uids` has a length greater than 0 before trying to access `Eids[i].Uids[0]` in order to avoid potential index out of bounds error. 

Another thing would be to check if `userExt.Eids[i].Uids[0].ID` is not an empty string before assigning to `unifiedId`

Also, these will be some good test cases to add ;)

Reply:
@mansinahar thanks for the feedback! 

Are you OK to merge this PR as-is (and I can create a new PR later with that change), or would you prefer if I make the changes tomorrow and add to this PR?

Reply:
@maphe We're planning on doing a release tomorrow so if you can make those changes by tomorrow, that's okay. I can then get it merged for it to be a part of that release.

Reply:
@mansinahar thank you! I've made the requested updates and pushed those out. Let me know if there is anything else we should add or change :)